packages/zipkin-transport-http/index.d.ts

```diff
@@ -1,12 +1,17 @@
+import {Agent as HttpAgent } from 'http';
+import {Agent as HttpsAgent} from 'https';
 import {JsonEncoder, Logger, model} from 'zipkin';
 
+type Agent = HttpAgent | HttpsAgent;
+
 declare class HttpLogger implements Logger {
   constructor(options: {
     endpoint: string,
     httpInterval?: number,
     jsonEncoder?: JsonEncoder,
     httpTimeout?: number,
     headers?: { [name: string]: any },
+    agent?: Agent | (() => Agent),
```

Comment:
is `(() => Agent)` the same effect as null?

Let's add typescript and javascript tests for this parameter to make sure someone later doesn't accidentally break what you need.

Reply:
`(() => Agent)` is the equivalent of a function which returns an Agent

This follows node-fetch which will accept an HTTP(S) Agent or a function which returns a HTTP(S) Agent